redux/redux.d.ts

```diff
@@ -1,50 +1,79 @@
-// Type definitions for Redux v1.0.0
+// Type definitions for Redux v3.0.0
 // Project: https://github.com/rackt/redux
-// Definitions by: William Buchwalter <https://github.com/wbuchwalter/>, Vincent Prouillet <https://github.com/Keats/>
+// Definitions by: William Buchwalter <https://github.com/wbuchwalter/>,
+//                 Vincent Prouillet <https://github.com/Keats/>,
+//                 Michael Tiller <https://github.com/xogeny/>
 // Definitions: https://github.com/borisyankov/DefinitelyTyped
 
 declare module Redux {
+    // Minimalist Action definition
+    interface Action {
+        type: string;
+    }
 
+    // Not used...?
     interface ActionCreator extends Function {
-        (...args: any[]): any;
+        (...args: any[]): Action;
+    }
+
+    // Also not used here...
+    interface StoreMethods<S,A extends Action> {
+        dispatch: Dispatch;
+        getState(): S;
     }
 
-    interface Reducer extends Function {
-        (state: any, action: any): any;
+    // Reducer is a function that takes a state of type S and an
+    // action and returns a new state, also of type S
+    interface Reducer<S> extends Function {
+        (state: S, action: any): S;
     }
 
+    // Dispatch is a function that takes an action of type A and
+    // returns an instance of that same action (still of type A)
     interface Dispatch extends Function {
-        (action: any): any;
+        <A extends Action>(action: A): any;
```

Comment:
This doesn't seem to be correct regarding the AsyncAction.
Ref : http://redux.js.org/docs/Glossary.html#dispatching-function